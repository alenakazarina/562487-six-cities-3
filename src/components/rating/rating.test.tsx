import * as React from 'react';
import * as renderer from 'react-test-renderer';
import Rating from './rating';

const RATINGS = [3, 5];
const PREFIXES = [`reviews`, `place-card`];

describe(`Rating`, () => {
  it(`should render review Rating with value, 3 stars, width = 60%`, () => {
    const tree = renderer.create(
        <Rating
          prefix={PREFIXES[0]}
          rating={RATINGS[0]}
          isValue
        />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it(`should render place card's Rating with 5 stars, width = 100%, without value`, () => {
    const tree = renderer.create(
        <Rating
          prefix={PREFIXES[1]}
          rating={RATINGS[1]}
          isValue={false}
        />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
