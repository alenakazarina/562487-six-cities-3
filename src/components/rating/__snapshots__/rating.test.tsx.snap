// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating should render place card's Rating with 5 stars, width = 100%, without value 1`] = `
<div
  className="place-card__rating rating"
>
  <div
    className="place-card__stars rating__stars"
  >
    <span
      style={
        Object {
          "width": "100%",
        }
      }
    />
    <span
      className="visually-hidden"
    >
      Rating
    </span>
  </div>
</div>
`;

exports[`Rating should render review Rating with value, 3 stars, width = 60% 1`] = `
<div
  className="reviews__rating rating"
>
  <div
    className="reviews__stars rating__stars"
  >
    <span
      style={
        Object {
          "width": "60%",
        }
      }
    />
    <span
      className="visually-hidden"
    >
      Rating
    </span>
  </div>
  <span
    className="reviews__rating-value rating__value"
  >
    3
  </span>
</div>
`;
